### PR TITLE
feat(store): warn on unhandled actions

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

**Comment by markwhitfeld** [comment]
@arturovt Could you add a description for this PR and the motivation for the feature?

File: packages/store/src/public_api.ts [hunk 1]
```diff
@@ -46,3 +46,7 @@ export { NgxsExecutionStrategy } from './execution/symbols';
 export { ActionType, ActionOptions } from './actions/symbols';
 export { NoopNgxsExecutionStrategy } from './execution/noop-ngxs-execution-strategy';
 export { StateToken } from './state-token/state-token';
+export {
+  UnhandledActionsLogger,
```
Comment (markwhitfeld) [inline packages/store/src/public_api.ts:50]: Is it necessary to export this type?
How would a user of the library use this, or when would they need to reference it?

File: packages/store/src/public_api.ts [hunk 2]
```diff
@@ -46,3 +46,7 @@ export { NgxsExecutionStrategy } from './execution/symbols';
...
+  ignoreActions
```
Comment (markwhitfeld) [inline packages/store/src/public_api.ts:51]: This `ignoreActions` API feels like it doesn't match the rest of the public API of the library.

File: packages/store/tests/unhandled-actions.spec.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  class FireAndForget {
+    static readonly type = 'Fire & forget';
+  }
+
+  const plainObjectAction = { type: 'Fire & Forget' };
+
+  const testSetup = (warnOnUnhandledActions = true) => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([], { warnOnUnhandledActions })]
+    });
```
Comment (markwhitfeld) [inline packages/store/tests/unhandled-actions.spec.ts:15]: I propose a bit of a different API for this.
No config option, or `ignoreActions` utility, but rather a module that is provided:
```suggestion
    TestBed.configureTestingModule({
      imports: [
        NgxsModule.forRoot([]),
        ngDevMode && NgxsDevelopmentModule.forRoot({ 
          warnOnUnhandledActions: true
          /* or supply an object:
            warnOnUnhandledActions: { 
              ignore: [ 'SomeActionType', FooActionClass ] 
            }
          */
        })
      ]
    });
```
This keeps the API consolidated and clear, and follows the existing conventions.

**Review by markwhitfeld — CHANGES_REQUESTED** [review]
I think that the API for this needs to be revised.
See the last comment.

File: packages/store/tests/unhandled-actions.spec.ts (hunk 3 above)
Comment (arturovt) [reply packages/store/tests/unhandled-actions.spec.ts:15]: Love it.

**Review by markwhitfeld — APPROVED** [review]

**Pull request merged**